src/Raven.Studio/typescript/viewmodels/database/settings/databaseSettings.ts

```diff
@@ -0,0 +1,444 @@
+                    
+                    if (rawEntry.Metadata.Scope === "ServerWideOnly") {
+                        return new models.serverWideOnlyEntry(rawEntry);
+                    }
+
+                    return models.databaseEntry.getEntry(rawEntry);
+                });
+
+                this.allEntries(_.sortBy(settingsEntries, x => x.keyName()));
+            });
+    }
+
+    private isEntriesValid() {
+        let valid = true;
+        let firstError = true;
+
+        this.allEntries().map(entry => {
+            if (this.getEntryStateClass(entry) === "customized-item-with-error") {
+                valid = false;
+
+                if (firstError) {
+                    firstError = false;
+                    this.setCategory(entry.data.Metadata.Category);
+                }
+            }
+        });
+
+        return valid;
+    }
+    
+    save() {
+        if (!this.isEntriesValid()) {
+            if (this.viewMode() === "summaryMode") {
+                this.switchToEditMode();
+            }
+
+            return;
+        }
+
+        const settingsToSave = this.allEntries().map(entry => {
+            if (entry instanceof models.databaseEntry) {
+                return entry.getEntrySetting();
+            }
+        }).filter(x => !!x);
+
+        const settingsToSaveSorted = _.sortBy(settingsToSave, x => x.key);
+        
+        const saveSettingsModel = new saveDatabaseSettingsConfirm((settingsToSaveSorted));
+
+        saveSettingsModel.result.done(result => {
+            if (result.can) {
+                this.spinners.save(true);
+
+                new saveDatabaseSettingsCommand(this.activeDatabase(), settingsToSaveSorted)
+                    .execute()
+                    .done(() => {
+                        this.fetchDatabaseSettings(this.activeDatabase())
+                            .done(() => this.processAfterFetch());
+                    })
+                    .always(() => this.spinners.save(false))
+                }
+            });
+
+        app.showBootstrapDialog(saveSettingsModel);
+    }
+
+    refresh() {
+        this.confirmRefresh()
+            .then(result => {
+                if (result.can) {
+                    this.spinners.refresh(true);
+
+                    this.fetchDatabaseSettings(this.activeDatabase(), true)
+                        .done(() => {
+                            this.processAfterFetch();
+                            this.dirtyFlag().reset();
+                        })
+                        .always(() => this.spinners.refresh(false));
+                }
+            });
+    }
+
+    private processAfterFetch() {
+        if (this.viewMode() === "editMode") {
+            this.setupEditMode();
+        } else {
+            this.computeEntriesToShow();
+        }
+    }
+    
+    confirmRefresh() {
+        let text = `<div>Clicking Refresh will reload all database settings from the server.</div>`;
```

Comment:
I wonder if we should dispaly confirmation only when we are dirty?

Reply:
done